roles/uperf/templates/service_metallb.yml.j2

```diff
@@ -82,6 +82,47 @@ items:
         targetPort: {{num}}
         protocol: UDP
 {% endfor %}
+{% if "sctp" in workload_args.protos %}
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name:  uperf-service-{{ item }}-{{ trunc_uuid }}-sctp
+      namespace: '{{ operator_namespace }}'
+      labels:
+        benchmark-uuid: {{ uuid }}
+        benchmark-operator-workload: uperf
+{% if (worker_node_list[node_idx_item]|length>27) and (worker_node_list[node_idx_item][26] == '.') %}
+        app: uperf-bench-server-{{ worker_node_list[node_idx_item] | truncate(26,true,'')}}-{{ item }}-{{ trunc_uuid }}
+{% else %}
+        app: uperf-bench-server-{{ worker_node_list[node_idx_item] | truncate(27,true,'')}}-{{ item }}-{{ trunc_uuid }}
+{% endif %}
+        type: {{ ansible_operator_meta.name }}-bench-server-{{ trunc_uuid }}-udp
+      annotations:
+        node_idx: '{{ node_idx_item }}'
+        pod_idx: '{{ item }}'
+        metallb.universe.tf/address-pool: '{{ workload_args.metallb.addresspool | default("addresspool-l2") }}'
+        metallb.universe.tf/allow-shared-ip: uperf-service-{{ item }}-{{ trunc_uuid }}
+    spec:
+      selector:
+{% if (worker_node_list[node_idx_item]|length>27) and (worker_node_list[node_idx_item][26] == '.') %}
+        app: uperf-bench-server-{{worker_node_list[node_idx_item] | truncate(26,true,'')}}-{{ item }}-{{ trunc_uuid }}
+{% else %}
+        app: uperf-bench-server-{{worker_node_list[node_idx_item] | truncate(27,true,'')}}-{{ item }}-{{ trunc_uuid }}
+{% endif %}
+      externalTrafficPolicy: '{{ workload_args.metallb.service_etp | default("Cluster") }}'
+      type: LoadBalancer
+      ports:
+      - name: uperf
+        port: 30000
+        targetPort: 30000
+        protocol: SCTP
+{% for num in range(30001,30012,1) %}
+      - name: uperf-control-tcp-{{num}}
```

Comment:
You mention `uperf-control-tcp` here, but have the protocol as SCTP? 

Is the proto sctp or tcp here? This is for the uperf control - not sure if it should be sctp or tcp... I haven't messed with sctp, so flying blind :(

Reply:
We are following the same naming convection for SCTP like the one used for TCP and UDP: https://github.com/cloud-bulldozer/benchmark-operator/blob/84d7f13ba2b0de41ad359ddb1c8b529f0a3ba1e2/roles/uperf/templates/service_metallb.yml.j2#L75-L83